src/iop/colorequal.c

```diff
@@ -277,8 +544,8 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const
 
     // Apply the corrections
     pix_out[0] += corrections_out[0]; // WARNING: hue is an offset
-    pix_out[1] *= corrections_out[1]; // the brightness and saturation are gains
-    pix_out[2] *= corrections_out[2];
+    pix_out[1] *= 1.f + corrections_out[1]; // the brightness and saturation are gains
+    pix_out[2] *= exp2f(2.f * corrections_out[2]);
```

Comment:
This needs to be

```c
pix_out[1] *= corrections_out[1];
pix_out[2] *= exp2f(2.f * (corrections_out[2] - 1.f));
```

The values from the sliders are in range [0, 2]. (Note that this doesn't fix the issues with the guided filter but at least the thing works without the guided filter with these)

Reply:
I personally find the `exp2f()` here kind of harsh especially when increasing the brightness. I wonder if it would be better to use the correction gain as is, i.e.

```c
pix_out[2] *= corrections_out[2];
```

But I also wondered if one should weight the brightness correction based on the saturation of the pixel. This way we would avoid doing excessive corrections of pixels that aren't actually colorful. A bit like what the color calibration module does for the colorfulness tab. This seems to give quite nice response:

```c
    // Apply the corrections
    pix_out[0] += corrections_out[1]; // WARNING: hue is an offset
    // Apply brightness and saturation gains.
    // Brightness correction is weighted based on the saturation of the pixel to
    // avoid introducing noise in near-achromatic parts of the image.
    pix_out[2] *= exp2f(10.f * pix_out[1] * (corrections_out[2] - 1.f));
    pix_out[1] *= corrections_out[0];
```